### ISSUE TITLE
Urgent: Timeouts not counting as failures in travis OR shippable

### ISSUE THREAD

**Issue by lavalamp** [GoogleCloudPlatform/kubernetes#7052]
Take great care merging!

Example PR #6989 run (look at the "succeeded" builds):

https://app.shippable.com/builds/553207173ef75c0c00a39b55

https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/59007463

**Comment by quinton-hoole** [comment]
Somewhat related, #7008 (bug) and #7029 (fix, as yet unmerged) mean that "guestbook" and "shell" e2e tests are being silently skipped on all platforms since Wed last week (4/15).  #7029 should merge shortly and resolve the issue.

**Comment by ixdy** [comment]
From the Travis run,
```
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/tools	300.128s
+++ [0418 07:38:41] Combined coverage report: /tmp/k8s_coverage/v1beta3/20150418-072840/combined-coverage.html
Job #16301.2
https://coveralls.io/jobs/5516029
The command "KUBE_RACE="-race" KUBE_COVER="y" KUBE_GOVERALLS_BIN="$HOME/gopath/bin/goveralls" KUBE_TIMEOUT='-timeout 300s' KUBE_COVERPROCS=8 KUBE_TEST_API_VERSIONS=$KUBE_TEST_API_VERSIONS ./hack/test-go.sh -- -p=2" exited with 0.
```

So it looks like either the shell script is eating the exit code, or the test is exiting 0 even when tests fail. Similarly for Shippable, it seems.

**Comment by ixdy** [comment]
I think #6899 broke this. It appears that `hack/test-go.sh` will *always* exit 0, even when tests fail or timeout. We should revert this change ASAP.

**Comment by lavalamp** [comment]
Fixed by #7058
